src/net_processing.cpp

```diff
@@ -1201,6 +1207,40 @@ void PeerLogicValidation::NewPoWValidBlock(const CBlockIndex *pindex, const std:
     });
 }
 
+static bool UpdateCFHeadersCache(const BlockFilterIndex& filter_index)
+{
+    LOCK(cs_main);
+    boost::unique_lock<boost::shared_mutex> _lock(active_chain_cf_headers_mtx);
+
+    CChain& active_chain = ::ChainActive();
```

Comment:
https://github.com/bitcoin/bitcoin/pull/16442/commits/3499e8877b3be1a629647353ddb8bb380c10a513

FWIW could make this `const`.